scripts/manpage_generation.sh

```diff
@@ -5,15 +5,41 @@ MANPAGE_DIR="man"
 
 echo Generating manpages
 
-VER=$(grep -oP '^Version:\s+\K\S+' packaging/convert2rhel.spec)
+# Detect changes in __init__.py
+if git diff --quiet HEAD~1 -- convert2rhel/__init__.py; then
+    # No changes detected, exit with 0
+    echo "No new version detected"
+    exit 0
+else
+    echo "enter changes deteced"
+    # Changes detected, generate manpages
 
-echo Generating for version $VER
-# Generate a file with convert2rhel synopsis for argparse-manpage
-/usr/bin/python -c 'from convert2rhel import toolopts; print("[synopsis]\n."+toolopts.CLI.usage())' > man/synopsis
+    #get dependcy
+    import subprocess
 
-/usr/bin/python -m pip install argparse-manpage six pexpect
+    # Install Python packages
+    subprocess.run(["pip", "install", "argparse-manpage", "six", "pexpect"])
 
-# Generate the manpage using argparse-manpage
-PYTHONPATH=. /usr/bin/python /home/runner/.local/bin/argparse-manpage --pyfile man/__init__.py --function get_parser --manual-title="General Commands Manual" --description="Automates the conversion of Red Hat Enterprise Linux derivative distributions to Red Hat Enterprise Linux." --project-name "convert2rhel $VER" --prog="convert2rhel" --include man/distribution --include man/synopsis > "$MANPAGE_DIR/convert2rhel.8"
 
-git status
+    # Generate a file with convert2rhel synopsis for argparse-manpage
+    /usr/bin/python -c 'from convert2rhel import toolopts; print("[synopsis]\n."+toolopts.CLI.usage())' > man/synopsis
+
+    /usr/bin/python -m pip install argparse-manpage six pexpect
+
+    # Extract the version of convert2rhel
+    VER=$(/usr/bin/python -c 'from convert2rhel import __version__; print(__version__)')
+
+    # Generate the manpage using argparse-manpage
+    PYTHONPATH=. /usr/bin/python /home/.local/bin/argparse-manpage --pyfile man/__init__.py --function get_parser --manual-title="General Commands Manual" --description="Automates the conversion of Red Hat Enterprise Linux derivative distributions to Red Hat Enterprise Linux." --project-name "convert2rhel $VER" --prog="convert2rhel" --include man/distribution --include man/synopsis > "$MANPAGE_DIR/convert2rhel.8"
+
+    # Check if manpages are up-to-date
+    if git diff --quiet HEAD -- "$MANPAGE_DIR/convert2rhel.8"; then
+        # Manpages are up-to-date, exit with 0
+        echo 'Manpages are up-to-date'
+        exit 0
+    else
```

Comment:
This is fine for now but we might need to modify this to support multiple versions soon, so we can support RHEL 9. Out of scope for this PR though

Reply:
Ok, should I make an Jira issue for that after this one is merged?

Reply:
I think we can leave it for now